src/pages/resultsView/oncoprint/AddClinicalTracks.tsx

```diff
@@ -17,16 +17,15 @@ import { ClinicalTrackConfig } from 'shared/components/oncoprint/Oncoprint';
 export interface IAddClinicalTrackProps {
     store: ResultsViewPageStore;
 
-    // We pass this in as a getter rather than passing the value itself because
-    //  mobx considers every prop changed if one prop changes, so it would do
-    //  unnecessary recomputation -> some rendering jitters whenever
-    //  selected clinical attributes would change. This way, that doesn't happen.
-    getSelectedClinicalAttributes: () => ClinicalTrackConfig[];
-
-    onChangeSelectedClinicalTracks: (
-        trackConfigs: ClinicalTrackConfig[]
-    ) => void;
-    clinicalTrackOptionsPromise: MobxPromise<{
+    clinicalAttributeIdToAvailableFrequencyPromise: MobxPromise<
+        ChartDataCountSet
+    >;
+    width?: number;
+    onAddAll: (keys: string[]) => void;
+    onClearAll: (keys: string[]) => void;
+    onToggleOption: (key: string) => void;
+    selectedClinicalAttributeIds: Dictionary<string>;
```

Comment:
see if you can use the Typescript Record type instead of this lodash supplied type.

Reply:
i mean, continue to use _.keyBy, that's correct, but see if you change the typing of this propery to Record if it will work.   Record is generic so you'll have to supply the key and value types, e.g. Record<string,string>